src/rgw/rgw_lc.cc

```diff
@@ -29,10 +29,6 @@
 #include "rgw_multi.h"
 #include "rgw_sal.h"
 
-// this seems safe to use, at least for now--arguably, we should
-// prefer header-only fmt, in general
-#undef FMT_HEADER_ONLY
-#define FMT_HEADER_ONLY 1
```

Comment:
the reason i requested this PR is because we were repeating this stuff in each translation unit that includes fmt. whether or not we decide to go with header-only, this decision belongs in cmake so it can be applied consistently across the project

Reply:
@cbodley so instead of hardwiring the cmake to `-DFMT_HEADER_ONLY`. 

1. i need to understand what does `FMT_HEADER_ONLY` offer to us? i asked for details, but no replies so far.
1. if we need it for some use cases.
  - we should have an option for it.
  - `fmt::fmt` does not only carry the library linkage, it also contains `INTERFACE_INCLUDE_DIRECTORIES`. instead of removing all occurrences of `fmt::fmt`, we could have a `fmt::header` library instead which exposes `-DFMT_HEADER_ONLY` and the `INTERFACE_INCLUDE_DIRECTORIES`.

Reply:
i think we should choose one and stick with it. if we provide both options, we're likely to end up with targets whose dependency chains include both. if there aren't good arguments for header-only, lets stick with static linkage